src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java

```diff
@@ -51,8 +53,28 @@ protected ResponseEntity handleInvalidTokenException() {
         return status(HttpStatus.UNAUTHORIZED).build();
     }
 
-    @ExceptionHandler(Exception.class)
+    @ExceptionHandler(ServiceBusException.class)
+    protected ResponseEntity handleServiceBusException() {
+        return status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .body("Server side exception occured while communicating with service bus");
+    }
+
+    @ExceptionHandler(InterruptedException.class)
+    protected ResponseEntity handleInterruptedException() {
+        return status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .body("Server side exception occurred as the thread was interrupted");
+    }
+
+    @ExceptionHandler(JsonProcessingException.class)
     protected ResponseEntity handleInternalException() {
+        return status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .body("Server side exception occured while parsing letter contents");
+    }
```

Comment:
will the client care?

Reply:
Well I have added that to the body as I found while working in divorce module while integrating with Document management that if you send 500 for different sometimes it is difficult to say why the request failed.Having message in the body we can determine in which scenario the request failed.But if you think it is not required I am happy to remove it.

Reply:
Perhaps we can skip the 'Server side exception occured while ' part thought. It's 500 after all.